### PR TITLE
C51-202: Fix color logic for due date

### PR DESCRIPTION
**Pull request by swastikpareek**
## Overview
 PR contains fix for the incorrect overdue styles on next activity card.

## Before

c51-202 - before

## After

c51-202 - after

## Technical Details

Checking the activity is overdue was not implemented generically throughout the app and hence was creating issues at some activities.
As there were three cases
* is_overdue was sent as Boolean for some activities
* is_overdue was sent as String for some activities
* is_overdue might not be present for some activities.

So to fix this we have a single function which will take care of this is used throughout the app now. At
* Activity Icon directive
* Case Card Variant
  * Case Card List view
  * Case Card Other cases view
* Activity Card Variant
  * Activity Card Big
  * Activity Card Short
  * Activity Card Long

### CODE REVIEW

File: ang/civicase/CaseDetails.js [hunk 1]
```diff
@@ -362,8 +362,6 @@
         crmApi('Case', 'getdetails', caseGetParams()).then(function (info) {
           $scope.pushCaseData(info.values[0]);
         });
-
-        $scope.item.nextActivityNotMilestone = findNextIncompleteActivityWhichIsNotMilestone($scope.item.allActivities); // to be removed after performace fix
```
Comment (deb1990) [inline ang/civicase/CaseDetails.js]: Not sure, Why is this removed?

File: ang/test/civicase/DateHelperService.spec.js [hunk 2]
```diff
@@ -18,16 +19,26 @@
 
       describe('isOverdue()', function () {
         beforeEach(function () {
-          pastDate = moment().subtract(7, 'd');
-          futureDate = moment().add(7, 'd');
+          activitiyWithBooleanOverdue = activitiesMockData[0];
```
Comment (deb1990) [inline ang/test/civicase/DateHelperService.spec.js]:
```javascript
describe('isOverdue()', function () {
  var activity;
  beforeEach(function () {
    activity = activitiesMockData[0];
  });

  describe('when api returns boolean value for is_overdue', function () {
    describe('when is_overdue is true', function () {
      beforeEach(function () {
        activity.is_overdue = true;
      });

      it('returns true if date is overdue',...)
    });

    describe('when is_overdue is false', function () {
      beforeEach(function () {
        activity.is_overdue = false;
      });

      it('returns false if date is not overdue',...)
    });
  });

  describe('when api returns string value for is_overdue', function () {
    describe('when is_overdue is true', function () {
      beforeEach(function () {
        activity.is_overdue = "1";
      });

      it('returns true if date is overdue',...)
    });

    describe('when is_overdue is false', function () {
      beforeEach(function () {
        activity.is_overdue = "0";
      });

      it('returns false if date is not overdue',...)
    });
  });
});
```

Comment (deb1990) [inline ang/test/civicase/DateHelperService.spec.js]: You are not checking the false values. Newly created variables should not be global. As it is used only inside `isOverdue` test.

File: ang/civicase/CaseDetails.js (hunk 1 above)
Comment (swastikpareek) [reply ang/civicase/CaseDetails.js]: > // to be removed after performance fix

This was the todo
